### PR TITLE
Feat/index prop

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
Description - added support for index prop in the SearchBox and SearchComponents for 
  - React-SearchBox
  - Vue-SearchBox
  - React-Native-SearchBox
  
  This prop is used to explicitly specify an index to query against for a particular component.
  
  Notion Link - https://www.notion.so/appbase/Index-key-FE-library-support-2255185cf3f94e0fb40de54050e4bbeb
  
  PR for Docs - 
    - React-SearchBox - https://github.com/appbaseio/Docs/pull/191
    - Vue-SearchBox - https://github.com/appbaseio/Docs/pull/190
    -React-Native-SearchBox - https://github.com/appbaseio/Docs/pull/192
    
  Demo - https://www.loom.com/share/0a96f4f6cbc044289224851b73a8a4e7

### CODE REVIEW

File: packages/vue-searchbox/src/index.js [hunk 1]
```diff
@@ -6,19 +6,19 @@ import version from './components/Version/index';
 const components = [SearchBox, SearchBase, SearchComponent];
 
 const install = function(Vue) {
```
Comment (bietkul) [inline packages/vue-searchbox/src/index.js]: @anik-ghosh-au7 Why did you change this file?

File: packages/react-searchbox/src/components/SearchComponent.js [hunk 2]
```diff
@@ -215,7 +215,7 @@ SearchComponent.defaultProps = {
 };
 
 SearchComponent.propTypes = {
-  index: string,
+  // index: string,
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchComponent.js]: @anik-ghosh-au7 Why did you change it?

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/vue-searchbox/src/index.js (hunk 1 above)
Comment (anik-ghosh-au7) [reply packages/vue-searchbox/src/index.js]: @bietkul reverted the changes

**Review by bietkul — APPROVED** [review]
